org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/util/JrtUtilTest.java

```diff
@@ -70,6 +74,20 @@ public void testGetNewJrtFileSystem() throws Exception {
 		assertSame(jrtSystem, jrtSystem3);
 	}
 
+	@Test
+	public void testWorksUnderInterruptCondition() throws IOException {
+		FileSystem fs = FileSystems.getFileSystem(URI.create("jrt:/"));
+		try {
+			Thread.currentThread().interrupt();
+			assertNotNull(JRTUtil.safeReadBytes(fs.getPath("modules", "java.base", "java/lang/Object.class")));
```

Comment:
this assertNotNull is the only condition that failed with the previous implementation

Reply:
> this assertNotNull is the only condition that failed with the previous implementation

That is correct and expected.